src/lib/util/api.ts

```diff
@@ -170,6 +170,23 @@ export function updateUserSetting<K extends keyof UserSettings>(
     });
 }
 
+export function changeUserPassword(oldPassword: string, newPassword: string, done?: () => void) {
+  const nid = notify({ type: "loading", text: "Changing Password" });
+  axios
+    .post("/user/change_password", { oldPassword, newPassword })
+    .then((r) => {
+      if (r.status === 200) {
+        notify({ id: nid, type: "success", text: "Password Changed" });
+        if (typeof done !== "undefined") done();
+      }
+    })
+    .catch((err) => {
+      const errMsg = err.response.data.error ? err.response.data.error : "Couldn't Change Password";
+      console.error("Change Password Form - Failed to change password on the server", err);
+      notify({ id: nid, type: "error", text: errMsg });
```

Comment:
Instead of using `notify` here, we could pass the error to the `done` callback. In the modal, we could use that to set `error` to show an error box at the top.

I think this should make it easier for users, so all errors show up in the same place.